src/test/java/com/google/sps/GivrUserTest.java

```diff
@@ -0,0 +1,145 @@
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.GivrUser;
+import com.google.sps.servlets.AddMaintainerServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.User;
+import com.google.gson.Gson;
+
+/** */
+@RunWith(JUnit4.class)
+public final class GivrUserTest {
+
+  private LocalServiceTestHelper helper;
+  private ArrayList<Entity> listOfEntities = new ArrayList<>();
+
+  private DatastoreService addEntitiesAndGetDatastore() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity user0 = new Entity("User");
+    user0.setProperty("userId", "User0");
+    user0.setProperty("isMaintainer", true);
+    user0.setProperty("userEmail", "jennb206+test0@gmail.com");
+    datastore.put(user0);
+    listOfEntities.add(user0);
+
+    Entity user1 = new Entity("User");
+    user1.setProperty("userId", "User1");
+    user1.setProperty("isMaintainer", false);
+    user1.setProperty("userEmail", "jennb206+test1@gmail.com");
+    datastore.put(user1);
+    listOfEntities.add(user1);
+
+    return datastore;
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void getUserFromDatastoreWithPropertyTest() {
+    helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+    helper.setUp();
+
+    DatastoreService datastore = addEntitiesAndGetDatastore();
+
+    String propertyName = "userEmail";
+    String propertyValue = "jennb206+test1@gmail.com";
+
+    Entity actualEntity = GivrUser.getUserFromDatastoreWithProperty(propertyName, propertyValue);
+    Entity expectedEntity = listOfEntities.get(1);
+
+    Assert.assertEquals(expectedEntity, actualEntity);
+  }
+
+  @Test
+  public void getCurrentLoggedInUserTestWhenNotLoggedIn() {
+    helper = new LocalServiceTestHelper(new LocalUserServiceTestConfig());
+    helper.setEnvIsLoggedIn(false);
+    helper.setUp();
+
+    GivrUser actualUser = GivrUser.getCurrentLoggedInUser();
+    GivrUser expectedUser = new GivrUser("", false, false, "logoutLinkHere.com", "");
+
+    Assert.assertEquals(expectedUser, actualUser);
+  }
+
+  @Test
+  public void getCurrentLoggedInUserTestWhenLoggedIn() {
+    LocalUserServiceTestConfig userServiceConfig = new LocalUserServiceTestConfig();
```

Comment:
It makes the tests a lot more useful for the future when you add a comment at the beginning of each one to explain exactly what is being tested for each one and what the expected behavior is